doc/getting-started.md

```diff
@@ -21,59 +19,72 @@ deployment "istio-mixer" created
+```
+kubectl expose deployment couchdb --name=couchdb
+# Let Istio know that the port is HTTP, not generic TCP
+kubectl patch service couchdb --type='json' -p='[{"op": "add", "path": "/spec/ports/0/name", "value":"http"}]'
+```
+
+Because the CouchDB service we created is not exposed to the outside world for security reasons
+we cannot test it from outside the cluster.  To verify that it is working correctly, we can run
+a _curl_ command against couchdb:5984 *from inside the cluster* using the public _dockerqa/curl_
+image from the Docker hub:
 
 ```
+kubectl run -i --rm --restart=Never dummy --image=dockerqa/curl:ubuntu-trusty --command -- curl couchdb:5984
+```
 
-<!---
-Note that to stand up the frontend without Istio, we can do
-kubectl run frontend --image=gcr.io/google-samples/hello-frontend:1.0 --port=8080 --overrides='{ "apiVersion": "extensions/v1beta1", "spec": { "containers": { "lifecycle": { "preStop": { "exec": { "command": ["/usr/sbin/nginx","-s","quit"] } } } } } }'
-# or
-kubectl run front-no-istio --image=gcr.io/google-samples/hello-frontend:1.0 --port=80
-# and then
-kubectl expose deployment front-no-istio --type=NodePort --name=frontend-no-istio
-kubectl describe services frontend-no-istio
--->
+The output should include '{"couchdb":"Welcome"...'
 
-At this point we have two microservices.  Let us test by looking up the public port
+Next we will start NGINX and configure it to proxy for the CouchDB service started in the previous step
 
 ```
-# Get the IP address.  (Alternate: gcloud compute instances list)
-minikube ip
-export IP=`minikube ip`
-# Get the NodePorts for both original and Istio versions
-kubectl describe services frontend
-export NODEPORT=...
+# Create an NGINX service
+kubectl run nginx --image=docker.io/nginx
+# The NGINX deployment must join the Istio service mesh.
+kubectl get deployment nginx --output yaml | istioctl kube-inject --filename - --tag 2017-03-22-17.30.06 | kubectl apply -f -
+# Configure the NGINX to point to the CouchDB.  Note that this step is just for the demo.  You should
+# not make changes to running containers in production.  This change is lost if the pod restarts.
+NGINX_POD=$(kubectl get pods -l run=nginx --output jsonpath='{.items[*].metadata.name}')
+kubectl exec -it $NGINX_POD /bin/bash -- -c \
+   "echo 'server { listen 9080; location / { proxy_pass http://couchdb:5984; proxy_http_version 1.1;  } }' > /etc/nginx/conf.d/frontend.conf && /usr/sbin/nginx -s reload"
+# (Note: The previous line should response with 'signal process started')
+# Expose the NGINX
+kubectl expose deployment nginx --type=NodePort --port 9080
 ```
 
-Test the service by doing `curl -i http://$IP:$NODEPORT`  For example, if the IP is 192.168.99.101 and NodePort is 31178/TCP, execute
+To test from outside the cluster we will need the IP address of a Kubernetes node, as well
+as the exposed NodePort port on that node.
```

Comment:
I suggest moving that curl to couchdb part here. And instead of querying couch db, query the frontend service via curl. 

Secondly, NodePort is not the only way to reach a service. There are other options (ingress, nodeport, loadbalancer). We need to clarify that to avail of routing functionality for user facing service, there needs to be an istio proxy (in the form of Ingress or standalone istio proxy service)